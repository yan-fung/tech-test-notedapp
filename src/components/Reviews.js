import { collection, getDocs } from "firebase/firestore";
import { db } from "../firebase";
import { useEffect } from "react";
import ReviewsCards from "./ReviewsCards";
import "../styles/reviews.scss";

const Reviews = ({ reviews, setReviews, isLoading, setIsLoading }) => {
  const fetchReview = async () => {
    setIsLoading(true);

    try {
      const querySnapshot = await getDocs(collection(db, "reviews"));
      const newData = querySnapshot.docs.map((doc) => ({
        ...doc.data(),
        id: doc.id,
      }));
      setReviews(newData);
      console.log(reviews, newData);
    } catch (error) {
      console.log("Error fetching data:", error);
    } finally {
      setIsLoading(false);
    }
  };

  useEffect(() => {
    fetchReview();
  }, []);

  // Get the first three reviews only
  const firstThreeReviews = reviews.slice(0, 3);

  return (
    <div className="review-content" data-testid="review__content">
      {isLoading ? (
        <div>Loading...</div>
      ) : (
        firstThreeReviews.map((review) => (
          <div key={review.id} className="item">
            <ReviewsCards
              body={review.body}
              reviewId={review.id}
              date={review.date}
              name={review.name}
              rating={review.rating}
              title={review.title}
            />
          </div>
        ))
      )}
    </div>
  );
};

export default Reviews;
