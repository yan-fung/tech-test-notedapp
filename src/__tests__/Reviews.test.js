import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Reviews from "../components/Reviews";

describe("Reviews component", () => {
  const testData = [1, 2, 3, 4, 5];
  const result = testData.slice(0, 3);

  it("renders the first three elements only", () => {
    expect(result).toEqual([1, 2, 3]);
  });

  // xit("render correctly", () => {
  //   const { asFragment } = render(
  //     <BrowserRouter>
  //       <Reviews result={result} />
  //     </BrowserRouter>
  //   );
  //   expect(asFragment()).toMatchSnapshot();
  // });
});
