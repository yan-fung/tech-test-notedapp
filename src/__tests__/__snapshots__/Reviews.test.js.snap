// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews component render correctly 1`] = `
<DocumentFragment>
  <div
    class="review-content"
    data-testid="review__content"
  >
    <div
      class="item"
    >
      <div>
        <div
          class="title"
        />
        <div
          class="rating"
          data-testid="review_rating"
        />
        <div
          class="date"
        />
        <div
          class="name"
        />
        <p
          class="body"
        />
      </div>
    </div>
    <div
      class="item"
    >
      <div>
        <div
          class="title"
        />
        <div
          class="rating"
          data-testid="review_rating"
        />
        <div
          class="date"
        />
        <div
          class="name"
        />
        <p
          class="body"
        />
      </div>
    </div>
    <div
      class="item"
    >
      <div>
        <div
          class="title"
        />
        <div
          class="rating"
          data-testid="review_rating"
        />
        <div
          class="date"
        />
        <div
          class="name"
        />
        <p
          class="body"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
